File: leetcode/archive/ex26.go
func removeDuplicates(nums []int) int {
    k, r := 0, 0
    for i := 1; i < len(nums); i++ {
        for k < len(nums)-1 && nums[k] == nums[k+1] {
            k += 1
        }
        if k < len(nums)-1 {
            nums[i] = nums[k+1]
            r = i
            k += 1
        }
    }
    return r+1
}

File: leetcode/ex26.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 1)
            return nums.size();
        int end = 0, pos = 1;
        while (end < nums.size()) {
            int begin = end;
            while (end < nums.size() && nums[end] == nums[begin])
                ++end;
            if (end < nums.size()) {
                nums[pos] = nums[end];
                ++pos;
            }
        }
        return pos;
    }
};
